docs/kind-example-config.yaml

```diff
@@ -3,8 +3,29 @@ apiVersion: kind.sigs.k8s.io/v1alpha1
 kind: Config
 # number of nodes in the cluster (currently only 1 is supported)
 numNodes: 1
-# template for kubeadm config, "" -> the default template
-kubeadmConfigTemplate: ""
+# these patches are applied to the generated kubeadm config
+# using `kustomize build` where the kustomization.yaml looks something like:
+# resources:
+# - generated-config.yaml
+# patches:
+# - file-containing-contents-of-the-first-patch.yaml
+# - file-containing-contents-of-the-second-patch.yaml
+# NOTE: this is somewhat brittle as the kubeadm config is not yet stable and
+# kind will need to select config versions relative to the kubernetes version
+# but it allows making advanced configuration when kind does not expose
+# any other options
+kubeadmConfigPatches:
+- |
+  apiVersion: kubeadm.k8s.io/v1alpha2
+  kind: MasterConfiguration
+  networking:
+    # Don't think setting pod subnet is currently required
+    # podSubnet: ""
+    serviceSubnet: 10.0.0.0/16
+  kubeletConfiguration:
+    baseConfig:
+      clusterDNS:
+      - 10.0.0.10
```

Comment:
Maybe add an example for `kubeadmConfigPatchesJson6902` as well.

Reply:
Good point, will do. Thanks :-)

Reply:
🤔 what is a `kubeadmConfigPatchesJson6902`? "Naming is hard" 😅

Reply:
Kustomize has field as `PatchesJson6902` to read in JSON patch. By the name, `kubeadmConfigPatchesJson6902` is added to for supporting JSON patch of kubeadminconfig.

Reply:
yeah I realized that this doesn't quite work though as the json patches also need a seperate non-inline GVK "Target" info. And it seems maybe we do still need to support these patches as strategic-merge-patch cannot neccesarily achieve the desired patch (eg appending to a list of string).

Reply:
example added after reworking.